src/Highlighter.ts

```diff
@@ -144,13 +144,34 @@ export default class Highlighter {
     return this.container.ownerDocument;
   }
 
+  private isIframe = (node: Node) => node.nodeName === 'IFRAME';
+
   private snapSelection = () => {
     const selection = this.document.getSelection();
 
     if (!selection || selection.isCollapsed) {
       return;
     }
 
+    const anchor = selection.anchorNode;
```

Comment:
lets move all this logic out of here and into the selection module under a function called like `cleanSelection` or something (that you can call here). please leave truly excessive comments about the purpose of things, like not just `// if selection begins on but doesn't end on iframe` but `... then do ___ because ___`